src/Views/Textile/Step.elm

```diff
@@ -594,80 +612,232 @@ stepHeader { current, inputs, toggleStep } =
-                                [ makingWasteField config
-                                , airTransportRatioField config
-                                , if inputs.product.making.fadable then
-                                    fadingField config
+                            Label.Making ->
+                                div [ class "mt-2" ]
+                                    [ makingWasteField config
+                                    , airTransportRatioField config
+                                    , if inputs.product.making.fadable then
+                                        fadingField config
 
-                                  else
-                                    text ""
-                                ]
+                                      else
+                                        text ""
+                                    ]
 
-                        Label.Use ->
-                            div [ class "mt-2" ]
-                                [ qualityField config
-                                , reparabilityField config
-                                , daysOfWearInfo inputs
-                                ]
+                            Label.Use ->
+                                div [ class "mt-2" ]
+                                    [ qualityField config
+                                    , reparabilityField config
+                                    , daysOfWearInfo inputs
+                                    ]
 
-                        _ ->
-                            text ""
+                            _ ->
+                                text ""
+                        ]
                     ]
-                , div [ class "col-sm-6 col-lg-5 text-center text-muted" ]
-                    [ div []
-                        [ if current.label /= Label.Distribution then
-                            div [ class "fs-3 fw-normal text-secondary" ]
-                                [ current.impacts
-                                    |> Impact.impactsWithComplements current.complementsImpacts
-                                    |> Format.formatImpact impact
+
+              else
+                viewMaterials config
+            ]
+    }
+
+
+viewMaterials : Config msg modal -> Html msg
+viewMaterials { db, current, inputs, impact, updateMaterial, deleteMaterial, setModal, addMaterialModal } =
+    ul [ class "CardList list-group list-group-flush" ]
+        (if List.isEmpty inputs.materials then
+            [ li [ class "list-group-item" ] [ text "Aucune matière première" ] ]
+
+         else
+            (inputs.materials
+                |> List.map
+                    (\materialInput ->
+                        let
+                            materialQuery : Inputs.MaterialQuery
+                            materialQuery =
+                                { id = materialInput.material.id
+                                , share = materialInput.share
```

Comment:
Nit: This `let…in` statement is huge, could we extract part of it into external functions, mostly for legibility purpose?

Reply:
Extracted most of it in a `createConfig` helper 👍